src/evalflags.cpp

```diff
@@ -98,7 +80,192 @@ EvalResult evalFlag(const FlagConfiguration& flag,
         event.variation = matchedSplit->variationKey;
         event.subject = subjectKey;
         event.subjectAttributes = subjectAttributes;
-        event.timestamp = toRFC3339(now);
+        event.timestamp = formatISOTimestamp(now);
+        event.metaData = {
+            {"sdkLanguage", "cpp"},
+            {"sdkVersion", SDK_VERSION}
+        };
+
+        // Convert extraLogging JSON to map of strings
+        if (matchedSplit->extraLogging.is_object()) {
+            for (auto& [key, value] : matchedSplit->extraLogging.items()) {
+                if (value.is_string()) {
+                    event.extraLogging[key] = value.get<std::string>();
+                } else {
+                    event.extraLogging[key] = value.dump();
+                }
+            }
+        }
+
+        result.event = event;
+    }
+
+    return result;
+}
+
+// Helper function to evaluate allocation with details
+AllocationEvaluationDetails evaluateAllocationWithDetails(
+    const Allocation& allocation,
+    const std::string& subjectKey,
+    const Attributes& augmentedSubjectAttributes,
+    int64_t totalShards,
+    const std::chrono::system_clock::time_point& now,
+    size_t orderPosition,
+    ApplicationLogger* logger) {
+
+    AllocationEvaluationDetails details;
+    details.key = allocation.key;
+    details.orderPosition = orderPosition;
+
+    // Check time constraints
+    if (allocation.startAt.has_value() && now < allocation.startAt.value()) {
+        details.allocationEvaluationCode = AllocationEvaluationCode::BEFORE_START_TIME;
+        return details;
+    }
+    if (allocation.endAt.has_value() && now > allocation.endAt.value()) {
+        details.allocationEvaluationCode = AllocationEvaluationCode::AFTER_END_TIME;
+        return details;
+    }
+
+    // Check if any rule matches
+    bool matchesRule = false;
+    if (!allocation.rules.empty()) {
+        for (const auto& rule : allocation.rules) {
+            if (ruleMatches(rule, augmentedSubjectAttributes, logger)) {
+                matchesRule = true;
+                break;
+            }
+        }
+        if (!matchesRule) {
+            details.allocationEvaluationCode = AllocationEvaluationCode::FAILING_RULE;
```

Comment:
`FAILING_RULE` also seems to be a confusing name since this applies to all rules (and rules are ORed). Maybe `ALL_RULES_FAILED` or `NO_RULES_MATCHED`?